src/code-considerations/using-unstable-lang/specialization.md

```diff
@@ -58,6 +58,39 @@ impl<T: Copy> RcFromSlice<T> for Rc<[T]> {
 
 Only specialization using the `min_specialization` feature should be used. The full `specialization` feature is known to be unsound.
 
+## Specialization attributes
+
+There are two unstable attributes that can be used to allow a trait bound in a specializing implementation that does not appear in the default implementation.
+
+`rustc_specialization_trait` restricts the implementations of a trait to be "always applicable". Implementing traits annotated with `rustc_specialization_trait` is unstable, so this should not be used on any stable traits exported from the standard library. `Sized` is an exception, and can have this attribute because it already cannot be implemented by an `impl` block.
```

Comment:
I would add a note that `rustc_specialization_trait` only prevents incorrect monomorphizations, it does not prevent a type from being coerced between an specialized and unspecialized type which can be important when specialization must be applied consistently. This edge-case lead to https://github.com/rust-lang/rust/issues/85863